### ISSUE TITLE
SaveData sends uninformative error messages when key column contains lists

### ISSUE THREAD

**Issue by zhizhongpu** [gslab-econ/gslab_python#180]
When specifying a key column that contains list elements, SaveData throws out uninformative error:
```
Traceback (most recent call last):
  File "/Users/zpu/Documents/GitHub/WorldOrder/source/derived/ISO/iso.py", line 140, in <module>
    Main()
  File "/Users/zpu/Documents/GitHub/WorldOrder/source/derived/ISO/iso.py", line 48, in Main
    SaveData(transitions, 
    ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zpu/Documents/GitHub/WorldOrder/source/lib/SaveData.py", line 14, in SaveData
    CheckKeys(df, keys)
  File "/Users/zpu/Documents/GitHub/WorldOrder/source/lib/SaveData.py", line 43, in CheckKeys
    if not all(df.groupby(keys).size() == 1):
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/groupby.py", line 3043, in size
    result = self._grouper.size()
             ^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/ops.py", line 705, in size
    ids, _, ngroups = self.group_info
                      ^^^^^^^^^^^^^^^
  File "properties.pyx", line 36, in pandas._libs.properties.CachedProperty.__get__
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/ops.py", line 745, in group_info
    comp_ids, obs_group_ids = self._get_compressed_codes()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/ops.py", line 764, in _get_compressed_codes
    group_index = get_group_index(self.codes, self.shape, sort=True, xnull=True)
                                  ^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/ops.py", line 690, in codes
    return [ping.codes for ping in self.groupings]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/ops.py", line 690, in <listcomp>
    return [ping.codes for ping in self.groupings]
            ^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/grouper.py", line 691, in codes
    return self._codes_and_uniques[0]
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "properties.pyx", line 36, in pandas._libs.properties.CachedProperty.__get__
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/groupby/grouper.py", line 835, in _codes_and_uniques
    codes, uniques = algorithms.factorize(  # type: ignore[assignment]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/algorithms.py", line 795, in factorize
    codes, uniques = factorize_array(
                     ^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/pt/lib/python3.11/site-packages/pandas/core/algorithms.py", line 595, in factorize_array
    uniques, codes = table.factorize(
                     ^^^^^^^^^^^^^^^^
  File "pandas/_libs/hashtable_class_helper.pxi", line 7281, in pandas._libs.hashtable.PyObjectHashTable.factorize
  File "pandas/_libs/hashtable_class_helper.pxi", line 7195, in pandas._libs.hashtable.PyObjectHashTable._unique
TypeError: unhashable type: 'list'
```

I see 2 options to fix this:
1. Explicitly ban lists from being used as keys; generate an error: lists detected in one of the keys
2. Allow keys columns to contain list elements.

**Comment by liaochris** [comment]
There should never be a scenario where lists are used as keys, especially because people can set a key to be a combination of multiple columns.

**Comment by liaochris** [comment]
In this issue, we added tests and functionality to inform the user that key column with any object of type list cannot be used

Deliverables are updated [`SaveData.py`](https://github.com/gslab-econ/gslab_python/blob/d5a03def1c43355022e00e3bdfb4d6e844609151/gslab_misc/SaveData/SaveData.py) and [`test_main.py`](https://github.com/gslab-econ/gslab_python/blob/d5a03def1c43355022e00e3bdfb4d6e844609151/gslab_misc/SaveData/tests/test_main.py)

Branch merged in 1524a57c2c48019ddc2af4b619045adc84a01725
Latest version of branch prior to merge: d5a03def1c43355022e00e3bdfb4d6e844609151
